src/implementationGuides/IGCompiler.ts

```diff
@@ -83,6 +90,23 @@ export class IGCompiler {
         return resources;
     }
 
+    async collectStructureDefinitions(igDir: PathLike): Promise<any[]> {
+        const indexJson = path.join(igDir.toString(), '.index.json');
+        if (!existsSync(indexJson)) {
+            throw new Error(`'.index.json' not found in ${igDir}`);
+        }
+        const index: any = await loadJson(indexJson);
+        const resources = [];
+        for (const file of index.files) {
+            if (file.resourceType === 'StructureDefinition') {
+                const filePath = path.join(igDir.toString(), file.filename);
+                console.log(`Compiling ${filePath}`);
+                resources.push(await loadJson(filePath));
+            }
+        }
+        return resources;
+    }
```

Comment:
same method as `collectSearchParams` except for line 101 & 74. Perhaps we can DRY it up with this method Signiture:
```ts
async collect(igDir: PathLike, type: 'SearchParameter'|'StructureDefinition'): Promise<any[]> {
```

Reply:
Done